[Pillar] Design guidelines (#283)
primer/brand: apps/docs/content/components/Pillar/index.mdx

---
title: Pillar
description: Use pillars to group related content together.
---

import ComponentLayout from '../../../src/layouts/component-layout'
export default ComponentLayout

## Anatomy

![An image showing the Pillar anatomy with an icon, a heading and a description in a vertical order.](https://github.com/primer/brand/assets/912236/c52debd2-4277-42c9-9a73-bd6bd087dd1e)

> - Icon: It provides a visual cue to the user about the content of the piller. Icon is optional.

Reviewer: ```suggestion
- Icon: An optional visual cue related to the content of the Pillar.
```

> - Heading: The main title of the pillar. It should be short and concise.

Reviewer: Should pillar be capitalized as it's the component name? We do this with all other components, like Grid below so I think we should do that here too.

```suggestion
- Heading: The main title of the Pillar. It should be short and concise.
```

- Description: Short text that extends the information provided by the heading.

## Usage

> Use pillars to group similar features or benefits together to easily scan and read. Pillars displays content in a familiar and recognizable style.

Reviewer 2: ```suggestion
Use pillars to group similar features or benefits together to easily scan and read. Pillars display content in a familiar and recognizable style.
```

<DoDontContainer>
  <Do>
    <img src="https://github.com/primer/brand/assets/912236/6fd15d68-91cc-446f-a813-c873aad9a8e7" />
    <Caption>
      Use a similar content size to achieve a visually balanced layout.
    </Caption>
  </Do>
  <Dont>
    <img src="https://github.com/primer/brand/assets/912236/4b33e84c-841a-43b1-a96b-5cec60ae0353" />
    <Caption>
      Don't use different length sizes for the title or the description.
    </Caption>
  </Dont>
</DoDontContainer>

If displaying an icon, do so for all cards to achieve a visually balanced layout.

<DoDontContainer>
  <Do>
    <img src="https://github.com/primer/brand/assets/912236/94dc144a-dffa-45a0-a77f-b0b830e669bc" />
    <Caption>
      Use icons either for all pillars or none for a visually balanced layout.
    </Caption>
  </Do>
  <Dont>
    <img src="https://github.com/primer/brand/assets/912236/8ddb7a7b-396b-44c2-9cbd-e6a44eb13710" />
    <Caption>Don't use icons sparingly.</Caption>
  </Dont>
</DoDontContainer>

> When presenting numerous pillars, we recommend stacking them vertically. You can use the [Grid](/components/Grid) or the [Stack](/components/Stack) component to achieve this. We recommend using a grid with 3 or 4 columns.

Reviewer 2: ```suggestion
When presenting more than four pillars, we recommend displaying them in multiple rows. You can use the [Grid](/components/Grid) or the [Stack](/components/Stack) component to achieve this. We recommend using a grid with 3 or 4 columns and balancing the number of pillars on each row.
```

![An image showing the Pillar anatomy with an icon, a heading and a description in a vertical order.](https://github.com/primer/brand/assets/912236/1d031ae9-bf4e-4e19-a7b7-c8cc0b7e5940)

## Options

### Align

> Pillars can be aligned to the start or center of the container. Aligning to the start is the default and recommended behavior. Use center alignment when in groups of 3 or 4 pillars and if their title and descriptions are short, so that the content is visually balanced and easy to scan.

Reviewer 2: I think we need a bit more guidance as to when center alignment can be used.
